bindings_node/src/enriched_message.rs

```diff
@@ -0,0 +1,155 @@
+  pub id: Vec<u8>,
+  pub sent_at_ns: i64,
+  pub kind: GroupMessageKind,
+  pub sender_installation_id: Vec<u8>,
+  pub sender_inbox_id: String,
+  pub content_type: ContentTypeId,
+  pub conversation_id: Vec<u8>,
+  pub fallback_text: Option<String>,
+  pub delivery_status: DeliveryStatus,
+  pub num_replies: i64,
+}
+
+#[napi]
+impl DecodedMessage {
+  // Lazy getter for reactions
+  #[napi(getter)]
+  pub fn reactions(&self) -> Vec<DecodedMessage> {
+    self
+      .inner
+      .reactions
+      .iter()
+      .map(|r| r.clone().into())
+      .collect()
+  }
+
+  // Lazy getter methods for content fields
+  #[napi(getter)]
+  pub fn text_content(&self) -> Option<TextContent> {
+    match &self.inner.content {
+      MessageBody::Text(t) => Some(t.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn reply_content(&self) -> Option<EnrichedReply> {
+    match &self.inner.content {
+      MessageBody::Reply(r) => Some(r.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn reaction_content(&self) -> Option<ReactionPayload> {
+    match &self.inner.content {
+      MessageBody::Reaction(r) => Some(r.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn attachment_content(&self) -> Option<Attachment> {
+    match &self.inner.content {
+      MessageBody::Attachment(a) => Some(a.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn remote_attachment_content(&self) -> Option<RemoteAttachment> {
+    match &self.inner.content {
+      MessageBody::RemoteAttachment(ra) => Some(ra.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn multi_remote_attachment_content(&self) -> Option<MultiRemoteAttachmentPayload> {
+    match &self.inner.content {
+      MessageBody::MultiRemoteAttachment(mra) => Some(mra.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn transaction_reference_content(&self) -> Option<TransactionReference> {
+    match &self.inner.content {
+      MessageBody::TransactionReference(tr) => Some(tr.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn group_updated_content(&self) -> Option<GroupUpdated> {
+    match &self.inner.content {
+      MessageBody::GroupUpdated(gu) => Some(gu.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn read_receipt_content(&self) -> Option<ReadReceipt> {
+    match &self.inner.content {
+      MessageBody::ReadReceipt(rr) => Some(rr.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn wallet_send_calls_content(&self) -> Option<WalletSendCalls> {
+    match &self.inner.content {
+      MessageBody::WalletSendCalls(wsc) => Some(wsc.clone().into()),
+      _ => None,
+    }
+  }
+
+  #[napi(getter)]
+  pub fn custom_content(&self) -> Option<EncodedContent> {
+    match &self.inner.content {
+      MessageBody::Custom(c) => Some(c.clone().into()),
+      _ => None,
+    }
+  }
+}
```

Comment:
curious, why all of the `*_content` methods versus a single `content` method?

Reply:
The FFI bindings return an enum that can be one of the content types. It doesn't seem like napi-rs supports complex enums. My plan is to turn all of these back into a union type at the layer above in JS.

Open to better ideas, since you've spent more time with napi than I have.

Reply:
ok, yea, that does seem to be a limitation of napi-rs. unfortunately, V3 doesn't seem to make this any better. cc @insipx in case we're missing something.

Reply:
the alternative if possible could be some trait that's boxed then passed over the barrier like an Object, but that would be a heavier lift for another time. also could be worth checking if the latest napi supports it, in which case we could update